nengo/utils/sockets.py

```diff
@@ -0,0 +1,322 @@
+            except socket.error:
+                pass
+            # Failed to open receiving socket, double backoff time, then retry
+            self.retry_backoff_time *= 2
+
+    def _terminate_alive_check_thread(self):
+        self.max_idle_time_current = self.max_idle_time_timeout
+        if self.alive_check_thread is not None:
+            self.last_active = 0
+            self.alive_check_thread.join()
+        self.alive_check_thread = None
+
+    def _close_send_socket(self):
+        if self.send_socket is not None:
+            self.send_socket.close()
+        self.send_socket = None
+
+    def _close_recv_socket(self):
+        if self.recv_socket is not None:
+            self.recv_socket.close()
+        self.recv_socket = None
+
+    def _config_wipe(self):
+        self.local_addr = '127.0.0.1'
+        self.local_port = -1
+        self.dest_addr = '127.0.0.1'
+        self.dest_port = -1
+        self.is_sender = False
+        self.is_receiver = False
+        self.ignore_timestamp = False
+
+        self.close()
+        self._initialize()
+
+    def config_send_only(self, dest_addr, dest_port):
+        self._config_wipe()
+        if (dest_port > 0):
+            self.is_sender = True
+            self.dest_addr = dest_addr
+            self.dest_port = dest_port
+        else:
+            raise ValueError("UDPSocket: Invalid send only configuration."
+                             "Destination port should be > 0")
+
+    def config_recv_only(self, local_port, timeout=5, ignore_timestamp=False):
+        self._config_wipe()
+        if (local_port > 0):
+            self.is_receiver = True
+            self.local_port = local_port
+            self.timeout = timeout
+            self.ignore_timestamp = ignore_timestamp
+        else:
+            raise ValueError("UDPSocket: Invalid recv only configuration."
+                             "Local port should be > 0")
+
+    def config_send_recv(self, local_port, dest_addr, dest_port, timeout=5,
+                         ignore_timestamp=False):
+        self._config_wipe()
+        if (local_port > 0 and dest_port > 0):
+            self.is_sender = True
+            self.is_receiver = True
+            self.dest_addr = dest_addr
+            self.dest_port = dest_port
+            self.local_port = local_port
+            self.timeout = timeout
+            self.ignore_timestamp = ignore_timestamp
+        else:
+            raise ValueError("UDPSocket: Invalid send and recv configuration."
+                             "Both destination and local ports should be > 0")
+
+    def set_byte_order(self, byte_order):
+        if (byte_order.lower() == "little"):
+            self.byte_order = '<'
+        elif (byte_order.lower() == "big"):
+            self.byte_order = '>'
+        else:
+            self.byte_order = byte_order
+
+    def pack_packet(self, t, x):
+        # pack_packet takes a timestamp and data (x) and makes a socket packet
```

Comment:
If you're going to make a multiline comment explaining how a function works, mark it with triple quotes.